src/App.js

```diff
@@ -1,10 +1,75 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
+import axios from 'axios';
+import styled from 'styled-components';
+import Title from './Title';
+import Candidate from './Candidate';
 
 function App() {
+  const [candidates, setCandidates] = useState([]);
+  const [rerenderTrigger, setRerenderTrigger] = useState(true);
+  let candidatesData = [];
+  let sortedCandidates = [];
+
+  useEffect(() => {
+    const getCandidate = async () => {
+      try {
+        const response = await axios.get(
+          'http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:2020/candidates'
+        );
+        candidatesData = response.data;
+      } catch (e) {
+        console.log('Candidates error : ', e);
+      }
+    }
+    
+    async function sortVotes() {
+      sortedCandidates = candidatesData
+        .sort((a, b) => {return b.voteCount - a.voteCount})
+    }
+
+    async function getAndSort() {
+      await getCandidate();
+      await sortVotes();
+      setCandidates(sortedCandidates);
+    }
+    getAndSort();
+  }, [rerenderTrigger]);
+
+  async function vote(cid, name) {
+    try {
+      await axios.get(
+        `http://ec2-3-34-5-220.ap-northeast-2.compute.amazonaws.com:2020/vote?id=${cid}`
+      );
+      await alert(`${name}님에게 투표 완료 !!!`);
+      await setRerenderTrigger(rerenderTrigger ? false : true);
+    } catch (error) {
+      alert('투표를 실패하셨어요...');
+      console.log('Vote error : ', error);
+    }
+  }
+
   return (
-    <div > 12기 프론트엔드 개발팀장 투표 ^.^
-    </div>
+    <Wrapper>
+      <Title />
+      <ul style={{listStyle: 'none', width: '100%'}}>
+        <li style={{width: '100%'}}>
+          {candidates.map((cand, i) => { return ( <Candidate name={cand.name} 
+          voteCount={cand.voteCount} cid={cand.id} rank={i} vote={vote} />)})}
+        </li>
+      </ul>
```

Comment:
1. inline 스타일링보다는` styled-component`를 사용하는 연습을해주세요ㅎㅎ귀찮으셨군요 창훈님 ㅠ..

2. 변수명은 생략하기보단 명확하게 적어주세요! `cand`->`candidate`가 더 좋겠죠??

3. `candidate.name`, `candidate.voteCount`, ... 넘겨주기 귀찮으니까 `candidate`자체를 넘겨서` Candidate.js`에서 비구조화할당으로 뽑아서 사용하는것도 괜찮을 것 같아요!

4.vote함수를 넘기려면 `updateVoteCount`로 보내서 사용해도 좋을 것 같아요

```suggestion
        return ( 
          <Candidate 
          {...{candidate}} 
          rank={i} 
          updateVoteCount={vote} />)})}
       
```

Reply:
아...!! candidate를 통째로 보내는 방법이 있었군요. ㅠㅠ 생각 못했네요
이 부분에선 inline 사용하는 편이 가독성 좋을 수도 있겠다 싶어서 한번 사용해 보았는데
완전히 스타일 컴포넌트로 통일시키는 편이 더 보기 좋나 보군용